bigtable/client/rpc_retry_policy_test.cc

```diff
@@ -13,65 +13,90 @@
   using namespace bigtable::chrono_literals;
-  bigtable::ExponentialBackoffPolicy tested(3, 10_ms, 150_ms);
-  
-  std::chrono::milliseconds delay;
-  EXPECT_TRUE(tested.on_failure(
-      grpc::Status(grpc::StatusCode::UNAVAILABLE, "please try again"), &delay));
-  EXPECT_TRUE(tested.on_failure(
-      grpc::Status(grpc::StatusCode::UNAVAILABLE, "please try again"), &delay));
-  EXPECT_TRUE(tested.on_failure(
-      grpc::Status(grpc::StatusCode::UNAVAILABLE, "please try again"), &delay));
-  EXPECT_FALSE(tested.on_failure(
-      grpc::Status(grpc::StatusCode::UNAVAILABLE, "please try again"), &delay));
+  bigtable::LimitedTimeRetryPolicy tested(50_ms);
+  check_limited_time(tested);
 }
 
-/**
- * @test Verify that non-retryable errors cause an immediate failure.
- */
-TEST(ExponentialBackoffRetryPolicy, OnNonRetryable) {
+/// @test Test cloning for LimitedTimeRetryPolicy.
+TEST(LimitedTimeRetryPolicy, Clone) {
   using namespace bigtable::chrono_literals;
-  bigtable::ExponentialBackoffPolicy tested(3, 10_ms, 150_ms);
+  bigtable::LimitedTimeRetryPolicy original(50_ms);
+  auto tested = original.clone();
+  check_limited_time(*tested);
+}
 
-  std::chrono::milliseconds delay;
-  EXPECT_FALSE(tested.on_failure(
-      grpc::Status(grpc::StatusCode::FAILED_PRECONDITION, "uh oh"), &delay));
+/// @test Verify that non-retryable errors cause an immediate failure.
+TEST(LimitedTimeRetryPolicy, OnNonRetryable) {
+  using namespace bigtable::chrono_literals;
+  bigtable::LimitedTimeRetryPolicy tested(10_ms);
+  EXPECT_FALSE(tested.on_failure(permanent()));
 }
 
-/**
- * @test Test cloning for ExponentialBackoffRetryPolicy.
- */
-TEST(ExponentialBackoffRetryPolicy, Clone) {
+/// @test A simple test for the LimitedErrorCountRetryPolicy.
+TEST(LimitedErrorCountRetryPolicy, Simple) {
   using namespace bigtable::chrono_literals;
-  bigtable::ExponentialBackoffPolicy original(3, 10_ms, 150_ms);
+  bigtable::LimitedErrorCountRetryPolicy tested(3);
+  check_limited_error_count(tested);
+}
+
+/// @test Test cloning for LimitedErrorCountRetryPolicy.
+TEST(LimitedErrorCountRetryPolicy, Clone) {
+  using namespace bigtable::chrono_literals;
+  bigtable::LimitedErrorCountRetryPolicy original(3);
   auto tested = original.clone();
+  check_limited_error_count(*tested);
+}
 
-  std::chrono::milliseconds delay;
-  EXPECT_TRUE(tested->on_failure(
-      grpc::Status(grpc::StatusCode::UNAVAILABLE, "please try again"), &delay));
-  EXPECT_TRUE(tested->on_failure(
-      grpc::Status(grpc::StatusCode::UNAVAILABLE, "please try again"), &delay));
-  EXPECT_TRUE(tested->on_failure(
-      grpc::Status(grpc::StatusCode::UNAVAILABLE, "please try again"), &delay));
-  EXPECT_FALSE(tested->on_failure(
-      grpc::Status(grpc::StatusCode::UNAVAILABLE, "please try again"), &delay));
+/// @test Verify that non-retryable errors cause an immediate failure.
+TEST(LimitedErrorCountRetryPolicy, OnNonRetryable) {
+  using namespace bigtable::chrono_literals;
```

Comment:
Looks like this is no longer needed in this scope.

Reply:
Thanks, fixed.